Maple2.Server.Game/PacketHandlers/FishingHandler.cs

```diff
@@ -0,0 +1,374 @@
+        }
+
+        if (session.Mastery[MasteryType.Fishing] < rodMetadata.MinMastery) {
+            session.Send(FishingPacket.Error(FishingError.s_fishing_error_fishingrod_mastery));
+            return;
+        }
+
+        if (!TableMetadata.FishingSpotTable.Entries.TryGetValue(session.Player.Value.Character.MapId, out FishingSpotTable.Entry? spotMetadata)) {
+            session.Send(FishingPacket.Error(FishingError.s_fishing_error_notexist_fish));
+            return;
+        }
+
+        if (session.Mastery[MasteryType.Fishing] < spotMetadata.MinMastery) {
+            session.Send(FishingPacket.Error(FishingError.s_fishing_error_lack_mastery));
+        }
+
+        IList<Vector3> fishingBlocks = GetFishingBlocks(session.Player.Position, session.Player.Rotation);
+
+        session.GuideObject = session.Field?.SpawnGuideObject(session.Player, new FishingGuideObject(), GetGuideObjectPosition(fishingBlocks, session.Player.Position));
+        session.GuideObject!.RodMetadata = rodMetadata;
+
+        session.Field?.Broadcast(GuideObjectPacket.Create(session.GuideObject));
+        session.Send(FishingPacket.LoadTiles(fishingBlocks));
+        session.Send(FishingPacket.Prepare(fishingRodUid));
+    }
+
+    private static Vector3 GetGuideObjectPosition(IEnumerable<Vector3> fishingBlocks, Vector3 playerPosition) {
+        // get the fishing block closest to player
+        Vector3 guidePosition = fishingBlocks.MinBy(o => Math.Sqrt(Math.Pow(playerPosition.X - o.X, 2) + Math.Pow(playerPosition.Y - o.Y, 2)));
+
+        // Move guide 1 block up so it sits on top of the fishing block
+        guidePosition.Z += 150;
+        return guidePosition;
+    }
+
+    private static IList<Vector3> GetFishingBlocks(Vector3 position, Vector3 rotation) {
+        /* Note this is all wrong and is a temp placeholder until the actual functionality is implemented.
+         Normally how it should work is we get the intercardinal direction that the player is facing closest to (Southwest, northwest, southeast, northeast)
+         We then scan in front of the player a 5x3x3 area for liquid blocks
+         The liquid block CANNOT have another block on top of it 
+         It cannot be ABOVE the block the player is standing on (i.e. scan the z axis same level as player and 2 blocks below max(?. May be more. Needs confirmation)
+         
+         For now, we will only be getting near-ish blocks. */
+
+        Vector3 roundedCoordinates = position;
+        
+        roundedCoordinates.X = (float) (Math.Round(roundedCoordinates.X / Constant.BlockSize) * Constant.BlockSize);
+        roundedCoordinates.Y = (float) (Math.Round(roundedCoordinates.Y / Constant.BlockSize) * Constant.BlockSize);
+        roundedCoordinates.Z = (float) (Math.Round(roundedCoordinates.Z / Constant.BlockSize) * Constant.BlockSize);
+
+        float direction = RotationDirectionUtil.GetClosestDirection(rotation);
+
+        var nearbyBlockCoordinates = new List<Vector3>();
```

Comment:
`VectorExtentions.BlocksInRange` might do what you want?

Reply:
Not exactly. It needs to a directional range of blocks